test/integration/api_v1_test.rb

```diff
@@ -608,6 +608,20 @@ class ApiV1Test < ActionDispatch::IntegrationTest
     assert_response :unauthorized
   end
 
+  test 'check if key exists' do
+    pid = @dessert_project.id
+    get "/api/v1/projects/#{pid}/key",
+        contribution_key: 'apple'
+    assert_response 302
+  end
+
+  test 'check if key exists (does not exist)' do
```

Comment:
rename this test to be less confusing. Also, make sure you have tests for all 4 conditions defined in the function.

1. bad params
2. no project
3. no key
4. success